create advanced search page

ErlendLandro opened the pull request:
I'm not sure if this new page should be a new page in the TOC or not. WDYT? 

It might be confusing to have two sources for search. I'm considering making the search icon in the top right corner link to this search page since it's more useful than the built-in search. 

Couldn't manage to get [these two](https://github.com/Starcounter/Docs/compare/2.1.177...error-page?expand=1#diff-f26dcab12df0ab18c5c541c64cbad1b2R24) event listeners to work in a loop to avoid code duplication. No idea why.

For Starcounter/TechComm#253

ErlendLandro committed:
create advanced search page

warpech commented on search/README.md:
```diff
@@ -0,0 +1,43 @@
+        return url.slice(questionMarkIndex + parameterLength + 1);
+    }
+    return "";
+}
```
The above function is a little bit too naive. It will fail when there are more parameters in the URL or if the URL contains special chars:

<img width="676" alt="screen shot 2017-04-21 at 17 01 46" src="https://cloud.githubusercontent.com/assets/566463/25283681/dd6e933e-26b4-11e7-81d1-c6ebbac3494e.png">

warpech commented on search/README.md:
```diff
@@ -0,0 +1,43 @@
+
+    frames[1].addEventListener("load", function() {
+        frames[1].contentDocument.getElementById("query").value = errorCode;
+    })
```
This is a little bit too complex. It would be totally fine (and more performant) to put these scripts inside the HTML files loaded in the iframes. I will commit a change that does that.

warpech reviewed (commented):
Good start. I have played a bit with improving this. Will push soon.

warpech committed:
search improvements

- change the query parameter name from "error" to "query" for more flexibility
- move JS code from content to the iframes (requires less "load" callbacks, improves performance)
- use a URL parsing function that unescapes special chars

warpech commented:
I've pushed a commit that proposes few improvements:

- change the query parameter name from "error" to "query" for more flexibility
- move JS code from content to the iframes (requires less "load" callbacks, improves performance)
- use a URL parsing function that unescapes special chars

WDYT?

ErlendLandro commented:
Oh wow! That's great. You certainly know these things better than I do. 

Is there anything else that should be done before merging?

warpech commented:
I think it is fine! Perhaps replace the search in the RebelsLounge GitBook with a link to this new page?

ErlendLandro commented:
> I think it is fine! Perhaps replace the search in the RebelsLounge GitBook with a link to this new page?

Great! I'll merge these changes and then replace the search in RebelsLounge with a link.

Thank you for the help!
